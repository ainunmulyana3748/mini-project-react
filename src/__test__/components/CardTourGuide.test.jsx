import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CardTourGuide from "../../components/CardTourGuide";

describe("CardTourGuide Component", () => {
  const user = {
    id: 1,
    avatar: "https://example.com/avatar.jpg",
    email: "tourguide@mail.com",
    first_name: "Budi",
    last_name: "Santoso",
  };

  test("render nama, email, dan avatar dengan benar", () => {
    render(
      <BrowserRouter>
        <CardTourGuide user={user} />
      </BrowserRouter>
    );
    expect(screen.getByText("Budi Santoso")).toBeInTheDocument();
    expect(screen.getByText("tourguide@mail.com")).toBeInTheDocument();
    const img = screen.getByAltText("Budi Santoso");
    expect(img).toHaveAttribute("src", user.avatar);
  });

  test("link menuju ke /users/{id}", () => {
    render(
      <BrowserRouter>
        <CardTourGuide user={user} />
      </BrowserRouter>
    );
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "users/1");
  });

  test("snapshot card tour guide", () => {
    const { container } = render(
      <BrowserRouter>
        <CardTourGuide user={user} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
