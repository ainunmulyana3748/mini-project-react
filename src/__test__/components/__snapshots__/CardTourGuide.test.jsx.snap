// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTourGuide Component snapshot card tour guide 1`] = `
<div>
  <a
    class="max-w-sm w-full bg-white rounded-2xl shadow-lg overflow-hidden cursor-pointer transform transition duration-300 hover:scale-105 hover:shadow-2xl"
    data-discover="true"
    href="/users/1"
  >
    <img
      alt="Budi Santoso"
      class="w-full h-48 object-cover"
      src="https://example.com/avatar.jpg"
    />
    <div
      class="p-6 text-center"
    >
      <h2
        class="text-xl font-semibold text-gray-800 group-hover:text-blue-600 transition-colors duration-200"
      >
        Budi
         
        Santoso
      </h2>
      <p
        class="text-sm text-gray-500 mt-1"
      >
        tourguide@mail.com
      </p>
    </div>
  </a>
</div>
`;
